artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/ServerSessionPacketHandler.java

```diff
@@ -921,7 +979,11 @@ private void doConfirmAndResponse(final Packet confirmPacket,
       }
 
       if (response != null) {
-         channel.send(response);
+         try {
+            channel.send(response);
```

Comment:
is this blocking or async, i ask, because if its async you need to be careful, putting back the object into the pool, as it may not be finished with.

Reply:
it's completely sync and won't retain the `Packet` unless the conf window size is != -1, where it can use a `resendCache` to retaining it.
In this case the caching won't happen, see https://github.com/apache/activemq-artemis/pull/3462/files#diff-c58d25d4978ebff7d03f1e5858b8b3344b01b3f2376e5ff194483d85c7b36a09R200-R208